src/plugins/discover/public/components/discover_grid/discover_grid_footer.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import classnames from 'classnames';
+import { EuiButtonEmpty } from '@elastic/eui';
+
+export interface DiscoverGridFooterProps {
+  isLoadingMore: boolean;
+  rowCount: number;
+  sampleSize: number;
+  pageIndex?: number;
+  pageCount: number;
+  totalHits?: number;
+  onFetchMoreRecords?: () => void;
+}
+
+export const DiscoverGridFooter: React.FC<DiscoverGridFooterProps> = (props) => {
+  const {
+    isLoadingMore,
+    rowCount,
+    sampleSize,
+    pageIndex,
+    pageCount,
+    totalHits = 0,
+    onFetchMoreRecords,
+  } = props;
+  const isOnLastPage = pageIndex === pageCount - 1 && rowCount < totalHits;
+
+  if (!isOnLastPage) {
+    return null;
+  }
+
+  // allow to fetch more records on Discover page
+  if (onFetchMoreRecords) {
+    // TODO: define a limit
+    if (rowCount <= 3000 - sampleSize) {
+      return (
+        <DiscoverGridFooterContainer hasButton={true} {...props}>
```

Comment:
Finding the right limit is a good question, it very much depends on the size of the records how much browsers resources / memory  fetching more data would take. dear @weltenwort, having more experience with incremental loading, what's the current limit of Logs, you recently mentioned it to me, thx!
Don't think we need to make this setting configurable until requested , but we should provide it in :
https://github.com/elastic/kibana/blob/92a46f53444c6e419967afa55b2e7664b7c9615a/src/plugins/discover/public/components/discover_grid/constants.ts#L14

Reply:
thx @weltenwort , resolving comment since it's old, and outdated